app/uk/gov/hmrc/soletraderidentification/controllers/JourneyDataController.scala

```diff
@@ -99,7 +103,7 @@ class JourneyDataController @Inject()(cc: ControllerComponents,
             _ => NoContent
```

Comment:
You could update this to throw an error if the repository method returns "false". This basically means there is no journey with the id provided. Have a look at how the existing code behaves. I think the method will execute without error if the journey id is ok, but there is no matching data field. However, if there is no journey data an error will be raised.